package/python-kiwi-keg.spec

```diff
@@ -77,7 +85,7 @@ more given git repositories that contain keg-recipes source tree. It supports
 auto-generation of change log files from commit history.
 
 %prep
-%setup -q -n keg-%{version}
+%setup -q -n kiwi_keg-%{version}
```

Comment:
Any specific reason for that change? This breaks when using the OBS source service. I used `keg` because that's what the `obs_scm` service creates when checking out the keg repository, and I couldn't find an easy way to change that directory name.

Reply:
Yes I had a specific reason and also wrote it in the commit message ;) ```keg``` is not the name of the python module. The name of the python module is ```kiwi_keg``` and the standard way to produce the dist target with the python build module automatically creates the source tarball with the correct names. As keg is also present on pypi this policy and actual requirement under certain conditions should not be violated.

I see you are using the recompress service in the _service definition and that creates you the tarball. But all of this is outside any standards and keg is also on pypi which should be considered the upstream. There is no correlation what the buildservice does here and all of this tar_scm stuff is a nightmare to maintain. Yes it makes your life easy as it might reduce explicit submission to obs but it hurts if you look at keg as a real opensource project with an upstream dist tarball on pypi, nothing matches than anymore

Regarding the obs service to build the package. I was pretty much confused what we are doing here. As said the source tarball does not follow python standards and the package build is a mix of data from ```_service``` plus some static files. The changes file does not correlate with the git history and the overall idea to build the package files is now a mix of scripts.

In addition no submission to the SUSE namespace will accept any ```_service``` file. Probably this is not a topic because Alex told me that we never wanted to have keg in SLES.

And the last part I'd like to mention, maybe I have forfeit my influence on keg when I left, but the idea for package maintenance on the keg project was as simple as this:

For upstream maintenance

```
bumpversion patch|minor|major
git tag --push
```

* github action triggers the pypi upload

For OBS maintenance

```
make build
cp dist/* /to/some/OBS/target
osc sr
```

* produces the exact same source tarball than upstream
* handles changes such that they are not out of sync with the git history
* is compatible with any submission target

I can imagine none of you is really interested in this sort of changes which I can understand as there is so much other stuff to do. But I strongly recommend to at least drop keg from pypi if the current package maintenance model should stay. If there is interest to maintain a healthy upstream, things needs to change.

just my2cents

Reply:
To make my point more clear. 

* .github/workflows/ci-publish-to-pypi.yml

```
- name: Prepare
      run: |
        python -m build
    - name: Publish
      uses: pypa/gh-action-pypi-publish@master
```

This is correct and the desired way to publish a python module to pypi

and in now obs

* Cloud:Tools/python-kiwi-keg

```
<services>
    <service name="obs_scm" mode="manual">
        <param name="filename">keg</param>

...

<service name="recompress" mode="manual">
    <param name="file">*.tar</param>
    <param name="compression">gz</param>
```
 
This is not the same and disconnected to upstream

Reply:
> Yes I had a specific reason and also wrote it in the commit message ;) `keg` is not the name of the python module. The name of the python module is `kiwi_keg` and the standard way to produce the dist target with the python build module automatically creates the source tarball with the correct names. As keg is also present on pypi this policy and actual requirement under certain conditions should not be violated.

Well, I never liked `kiwi_keg` to start with, because the k in `keg` already stands for `kiwi`. ;)

The package is named `python-kiwi-keg` and thus complies with the naming convention. I don't really consider the name of the automatically generated tarball which is only used in the build service subject to that naming convention, but I guess it's debatable.

> I see you are using the recompress service in the _service definition and that creates you the tarball. But all of this is outside any standards and keg is also on pypi which should be considered the upstream. There is no correlation what the buildservice does here and all of this tar_scm stuff is a nightmare to maintain. Yes it makes your life easy as it might reduce explicit submission to obs but it hurts if you look at keg as a real opensource project with an upstream dist tarball on pypi, nothing matches than anymore

Using `obs_scm` and `recompress` seems pretty common to me in the build service. I don't consider pypi to be upstream. The github repo is the source of truth and pypi is a distribution, in a way similar to what OBS produces.

> Regarding the obs service to build the package. I was pretty much confused what we are doing here. As said the source tarball does not follow python standards and the package build is a mix of data from `_service` plus some static files. The changes file does not correlate with the git history and the overall idea to build the package files is now a mix of scripts.

I don't follow. Everything is generated from source including the change log.

> In addition no submission to the SUSE namespace will accept any `_service` file. Probably this is not a topic because Alex told me that we never wanted to have keg in SLES.

That's not correct. Server-side service must not be used in in maintained IBS projects.  Manual services are fine and there are `_service` files in IBS.

> And the last part I'd like to mention, maybe I have forfeit my influence on keg when I left, but the idea for package maintenance on the keg project was as simple as this:

You didn't forfeit anything. :)

> For upstream maintenance
> 
> ```
> bumpversion patch|minor|major
> git tag --push
> ```
> 
> * github action triggers the pypi upload
> 
> For OBS maintenance
> 
> ```
> make build
> cp dist/* /to/some/OBS/target
> osc sr
> ```
> 
> * produces the exact same source tarball than upstream

Debatable. :) It'll happily pick up local changes. Generally I'm not a fan of packaging up what's on a developers machine. Directly generating from source makes sure it's clean.

> * handles changes such that they are not out of sync with the git history
> * is compatible with any submission target
> 
> I can imagine none of you is really interested in this sort of changes which I can understand as there is so much other stuff to do. On the other side I believe upstream sources and what suse builds should be the same

As said, I consider the git repo to be the upstream source and what the src service generates is equivalent to the source tarball you can download from github. Said that, I don't really care. If having the same source tarball in OBS as in pypi is important for some reason we can switch methods.

Reply:
> Well, I never liked kiwi_keg to start with, because the k in keg already stands for kiwi

It was named that way because "keg" was already taken on pypi. 

> The package is named python-kiwi-keg and thus complies with the naming convention. 

This is not true and a typical "SUSE we know better answer" sorry to say so but the upstream for kiwi-keg is pypi not github. People who would package it for other distros would take it from pypi because it releases there with a version. The namespace for keg is kiwi_keg as you can see it in any of your imports and a tarball for a python project should be created using python build utilities and not some "SUSE recompress service"

> I don't consider pypi to be upstream

That says everything, so why do we have it there ?